### PR TITLE
fix: update eslint to @edx/eslint-config

### PR DESCRIPTION
**Pull request by raq929**
Autofixes most issues
Ignores componentWillReceiveProps and import/no-cycle errors

### CODE REVIEW

**Comment by abutterworth** [comment]
@raq929 It looks like snapshot tests are failing.

**Comment by raq929** [comment]
Yeah, sorry, I made a commit to fix that and then forgot to push it.

File: package.json [hunk 1]
```diff
@@ -59,24 +59,30 @@
...
     "coveralls": "^3.0.3",
     "enzyme": "^3.9.0",
     "enzyme-adapter-react-16": "^1.11.2",
-    "eslint-config-edx": "^4.0.4",
+    "eslint": "^6.8.0",
+    "eslint-config-airbnb": "^18.2.0",
+    "eslint-plugin-import": "^2.22.0",
+    "eslint-plugin-jsx-a11y": "^6.3.1",
+    "eslint-plugin-react": "^7.20.6",
+    "eslint-plugin-react-hooks": "^1.7.0",
```
Comment (abutterworth) [inline package.json:74]: Why is this needed?

File: src/Dropdown/deprecated/index.jsx [hunk 2]
```diff
@@ -163,39 +164,37 @@ Dropdown.propTypes = {
...
 
-
 const DropdownWithDeprecatedProps = withDeprecatedProps(Dropdown, 'Dropdown', {
   menuItems: {
     deprType: DEPR_TYPES.MOVED_AND_FORMAT,
     message: 'They should be components sent as children.',
     newName: 'children',
     transform: (menuItems, allProps) => {
-      if (!Array.isArray(menuItems)) return null;
+      if (!Array.isArray(menuItems)) { return null; }
```
Comment (abutterworth) [inline src/Dropdown/deprecated/index.jsx:177]: This is functionally different. I'd add an linter ignore comment here instead

File: package.json (hunk 1 above)
Comment (raq929) [reply package.json:74]: I followed the directions here: https://github.com/edx/eslint-config
(Namely `npx install-peerdeps --dev @edx/eslint-config`), and this is everything it installed.

**Comment by raq929** [comment]
@abutterworth In my email I'm seeing a comment to change a functional difference into a linter ignore comment; but I'm not seeing that here. Did you change your mind on that, or is github being flaky?

**Comment by abutterworth** [comment]
> @abutterworth In my email I'm seeing a comment to change a functional difference into a linter ignore comment; but I'm not seeing that here. Did you change your mind on that, or is github being flaky?

I deleted my comment because it was wrong :P. finishing up the review now
